src/lib/adapters/firebase/index.ts

```diff
@@ -373,11 +379,34 @@ export default class FirebaseAdapter implements Adapter {
 		// Calculate and deduct fee
 		const feeToken = (await getDoc(feeDoc)).data()
 		if (!feeToken) throw new Error('Fee not found')
-		setDoc(feeDoc, { amount: (BigInt(feeToken.amount) - fee.amount).toString() }, { merge: true })
+		setDoc(feeDoc, { amount: (BigInt(feeToken.amount) - fee.amount).toString() }, { merge: true })	
+		*/
+
+		const tx = await sendTransaction(wallet, to, token.amount, fee.amount)
+
+		await this.updateBalance(wallet.address, token)
 
 		const txCollection = collection(db, `transactions`)
-		const res = await addDoc(txCollection, tx)
-		return res.id
+		const txData = {
+			from: address,
+			to,
+			token: {
+				amount: token.amount.toString(),
+				name: token.name,
+				symbol: token.symbol,
+				decimals: token.decimals,
+			},
+			timestamp: Date.now(),
+			fee: {
+				amount: fee.amount.toString(),
+				symbol: fee.symbol,
+				name: fee.name,
+				decimals: fee.decimals,
+			},
+		}
+
+		const res = await addDoc(txCollection, txData)
```

Comment:
Why are we storing this? And why is `res` assigned but not used?

Reply:
I opened an issue about this topic: https://github.com/logos-innovation-lab/waku-objects-playground/issues/171